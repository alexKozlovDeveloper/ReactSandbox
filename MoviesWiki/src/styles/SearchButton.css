.button {
    background-color: #222327;
    border: none;
    color: white;
    padding: 6px 10px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    cursor: pointer;
    margin-left: 5px;
}