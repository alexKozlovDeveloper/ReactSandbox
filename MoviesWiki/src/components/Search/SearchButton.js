import React, { Component } from "react";

import styles from "../../styles/SearchButton.css"

const SearchButton = ({ buttonText, updateFunc }) => {
    return  <button className={styles.button} onClick={() => updateFunc()}>{buttonText}</button>;
}

export default SearchButton;