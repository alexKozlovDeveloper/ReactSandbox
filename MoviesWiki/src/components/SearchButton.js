import React, { Component } from "react";

import styles from "../styles/SearchButton.css"

function SearchButton(props) {
    return  <button className={styles.button}>{props.buttonText}</button>;
}

export default SearchButton;