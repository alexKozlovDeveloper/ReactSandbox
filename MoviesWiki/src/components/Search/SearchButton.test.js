import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import SearchButton from "./SearchButton";

configure({adapter: new Adapter()});

test('SearchButton rendering test', () => {

    var buttonText = 'some text';

    const component = shallow(<SearchButton buttonText={buttonText}/>);
    expect(shallowToJson(component)).toMatchSnapshot();
 });

