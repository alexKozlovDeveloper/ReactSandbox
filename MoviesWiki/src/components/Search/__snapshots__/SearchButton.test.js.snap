// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton rendering test 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  some text
</button>
`;
